src/csync/csync_private.h

```diff
@@ -67,14 +67,47 @@ enum csync_replica_e {
   REMOTE_REPLICA
 };
 
+
+/*
+ * This is a structurere similar to QStringRef
+ * The difference is that it keeps the QByteArray by value and not by pointer
+ * And it only implements a very small subset of the API that is required by csync, the API can be
+ * added as we need it.
+ */
+class ByteArrayRef
```

Comment:
Did you just optimize QByteArray and it immedietaly gives 1/4 speed-up for csync_reconcile? nice :+1:

Reply:
Instead of using QByteArray, i use a structure that is a slice of a QByteArray. Before we created lots of QByteArray for the parent directory, but this allocated and copied a lot of memory.  By just having a slice, we don't need to allocate.